src/protocols/bdx/BdxMessages.cpp

```diff
@@ -0,0 +1,401 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      Implements utility methods for working with some complex BDX messages.
+ */
+
+#include <protocols/bdx/BdxMessages.h>
+
+#include <support/BufBound.h>
+#include <support/BufferReader.h>
+#include <support/CodeUtils.h>
+
+#include <limits>
+
+namespace {
+constexpr uint8_t kVersionMask = 0x0F;
+} // namespace
+
+using namespace chip;
+using namespace chip::BDX;
+using namespace chip::Encoding::LittleEndian;
+
+CHIP_ERROR TransferInit::Pack(System::PacketBuffer & aBuffer) const
+{
+    CHIP_ERROR err              = CHIP_NO_ERROR;
+    uint8_t proposedTransferCtl = 0;
+    bool widerange              = ((mStartOffset | mMaxLength) > std::numeric_limits<uint32_t>::max());
+    BufBound bbuf(aBuffer.Start(), aBuffer.AvailableDataLength());
+
+    proposedTransferCtl |= mSupportedVersions & kVersionMask;
```

Comment:
Seeing how `mSupportedVersions` is used, `mSupportedVersion` (not plural) is the correct naming.